### PR TITLE
Rename 'activity' to 'element' in all records

### LINKED ISSUE #1528: Workflow instance record: Rename activity id to element id
We use this field to store the IDs of activities, sequence flows, and the process. Their closest common ancestor in the BPMN meta-model is `BaseElement`, so `elementId` is more accurate than `activityId`.

### PR DESCRIPTION
**Pull request by ThorbenLindhauer**
- workflow instance record
- incident record
- job headers
- message subscription
- workflow subscription

closes #1528

### CODE REVIEW

**Comment by menski** [comment]
The elasticsearch exporter mappings have to be adjusted to the changes, see https://github.com/zeebe-io/zeebe/tree/1528-element-id/exporters/elasticsearch-exporter/src/main/resources

**Comment by menski** [comment]
Reminder we have to regenerate the Go gRPC stub and mocks as this is not done automatically. See for instructions:
- https://github.com/zeebe-io/zeebe/blob/develop/gateway-protocol/README.md
- https://github.com/zeebe-io/zeebe/blob/develop/clients/go/README.md

In case you need help or want me to do it just ping me

File: broker-core/src/main/java/io/zeebe/broker/workflow/processor/job/JobCompletedEventProcessor.java [hunk 1]
```diff
@@ -44,7 +44,7 @@ public void processRecord(
 
     final JobRecord jobEvent = record.getValue();
     final JobHeaders jobHeaders = jobEvent.getHeaders();
-    final long activityInstanceKey = jobHeaders.getActivityInstanceKey();
+    final long activityInstanceKey = jobHeaders.getElementInstanceKey();
```
Comment (saig0) [inline broker-core/src/main/java/io/zeebe/broker/workflow/processor/job/JobCompletedEventProcessor.java]:
```suggestion
    final long elementInstanceKey = jobHeaders.getElementInstanceKey();
```

File: broker-core/src/main/java/io/zeebe/broker/workflow/processor/job/JobCreatedProcessor.java [hunk 2]
```diff
@@ -41,7 +41,7 @@ public void processRecord(
       TypedStreamWriter streamWriter) {
 
     final JobHeaders jobHeaders = record.getValue().getHeaders();
-    final long activityInstanceKey = jobHeaders.getActivityInstanceKey();
+    final long activityInstanceKey = jobHeaders.getElementInstanceKey();
```
Comment (saig0) [inline broker-core/src/main/java/io/zeebe/broker/workflow/processor/job/JobCreatedProcessor.java]:
```suggestion
    final long elementInstanceKey = jobHeaders.getElementInstanceKey();
```

File: broker-core/src/main/java/io/zeebe/broker/workflow/processor/servicetask/TerminateServiceTaskHandler.java [hunk 3]
```diff
@@ -50,8 +50,8 @@ protected void addTerminatingRecords(
           .setBpmnProcessId(activityInstanceEvent.getBpmnProcessId())
           .setWorkflowDefinitionVersion(activityInstanceEvent.getVersion())
           .setWorkflowInstanceKey(activityInstanceEvent.getWorkflowInstanceKey())
-          .setActivityId(activityInstanceEvent.getActivityId())
-          .setActivityInstanceKey(activityInstance.getKey());
+          .setElementId(activityInstanceEvent.getElementId())
```
Comment (saig0) [inline broker-core/src/main/java/io/zeebe/broker/workflow/processor/servicetask/TerminateServiceTaskHandler.java]:
```suggestion
          .setElementId(elementInstanceEvent.getElementId())
```

File: clients/java/src/main/java/io/zeebe/client/api/response/JobHeaders.java [hunk 4]
```diff
@@ -31,8 +31,8 @@
   long getWorkflowKey();
 
   /** @return id of the activity */
```
Comment (saig0) [inline clients/java/src/main/java/io/zeebe/client/api/response/JobHeaders.java]:
```suggestion
  /** @return id of the element */
```

File: clients/java/src/main/java/io/zeebe/client/api/response/JobHeaders.java [hunk 5]
```diff
@@ -31,8 +31,8 @@
...
-  String getActivityId();
+  String getElementId();
 
   /** @return key of the activity instance */
```
Comment (saig0) [inline clients/java/src/main/java/io/zeebe/client/api/response/JobHeaders.java]:
```suggestion
  /** @return key of the element instance */
```

File: gateway/src/main/java/io/zeebe/gateway/api/record/IncidentRecord.java [hunk 6]
```diff
@@ -38,13 +38,13 @@
    * @return the id of the activity this incident belongs to. Can be <code>null</code> if the
```
Comment (saig0) [inline gateway/src/main/java/io/zeebe/gateway/api/record/IncidentRecord.java]:
```suggestion
   * @return the id of the element this incident belongs to. Can be <code>null</code> if the
```

File: gateway/src/main/java/io/zeebe/gateway/api/record/IncidentRecord.java [hunk 7]
```diff
@@ -38,13 +38,13 @@
...
    *     incident belongs to no activity or workflow instance.
    */
-  String getActivityId();
+  String getElementId();
 
   /**
    * @return the key of the activity instance this incident belongs to. Can be <code>null</code> if
```
Comment (saig0) [inline gateway/src/main/java/io/zeebe/gateway/api/record/IncidentRecord.java]:
```suggestion
   * @return the key of the element instance this incident belongs to. Can be <code>null</code> if
```

File: gateway/src/main/java/io/zeebe/gateway/api/record/WorkflowInstanceRecord.java [hunk 8]
```diff
@@ -39,7 +39,7 @@
   /**
    * @return the id of the current activity, or empty if the event does not belong to an activity.
```
Comment (saig0) [inline gateway/src/main/java/io/zeebe/gateway/api/record/WorkflowInstanceRecord.java]:
```suggestion
   * @return the id of the current element, or empty if the event does not belong to an activity.
```

File: gateway/src/main/java/io/zeebe/gateway/impl/command/IncidentCommandImpl.java [hunk 9]
```diff
@@ -50,9 +50,9 @@ public String toString() {
     builder.append(", workflowInstanceKey=");
     builder.append(getWorkflowInstanceKey());
     builder.append(", activityId=");
```
Comment (saig0) [inline gateway/src/main/java/io/zeebe/gateway/impl/command/IncidentCommandImpl.java]:
```suggestion
    builder.append(", elementId=");
```

File: gateway/src/main/java/io/zeebe/gateway/impl/command/WorkflowInstanceCommandImpl.java [hunk 10]
```diff
@@ -63,7 +63,7 @@ public String toString() {
     builder.append(", version=");
     builder.append(getVersion());
     builder.append(", activityId=");
```
Comment (saig0) [inline gateway/src/main/java/io/zeebe/gateway/impl/command/WorkflowInstanceCommandImpl.java]:
```suggestion
    builder.append(", elementId=");
```

File: gateway/src/main/java/io/zeebe/gateway/impl/event/IncidentEventImpl.java [hunk 11]
```diff
@@ -50,9 +50,9 @@ public String toString() {
     builder.append(", workflowInstanceKey=");
     builder.append(getWorkflowInstanceKey());
     builder.append(", activityId=");
```
Comment (saig0) [inline gateway/src/main/java/io/zeebe/gateway/impl/event/IncidentEventImpl.java]:
```suggestion
    builder.append(", elementId=");
```

File: gateway/src/main/java/io/zeebe/gateway/impl/event/WorkflowInstanceEventImpl.java [hunk 12]
```diff
@@ -51,7 +51,7 @@ public String toString() {
     builder.append(", version=");
     builder.append(getVersion());
     builder.append(", activityId=");
```
Comment (saig0) [inline gateway/src/main/java/io/zeebe/gateway/impl/event/WorkflowInstanceEventImpl.java]:
```suggestion
    builder.append(", elementId=");
```

**Review by saig0 — CHANGES_REQUESTED** [review]

**Comment by GitCop** [comment]
There were the following issues with your Pull Request

 * Commit: 5833cebad35ae1691e29dffc5cd7991f4cdac7fa
  * Commits must be in the following format: **%{type}(%{scope}): %{description}**

Guidelines are available at https://github.com/zeebe-io/zeebe/blob/develop/CONTRIBUTING.md

---
This message was auto-generated by https://gitcop.com

**Comment by ThorbenLindhauer** [comment]
@saig0

I have made the following changes:

- updated the locations that you marked
- re-generated the go client and adapted where necessary
- updated the debug http exporter (was @menski actually :)

**Review by saig0 — APPROVED** [review]

**Pull request merged**